docs/source/architecture.rst

```diff
@@ -4,14 +4,62 @@
 Architecture of nbconvert
 =========================
 
-This is a high-level outline of the structure and objects in nbconvert,
-and how they are used in the pipeline of converting a notebook to any given format.
-
+This is a high-level outline of the basic workflow, structures and objects in nbconvert.
+Specifically, this exposition has a two-fold goal: 
+    
+    #. to alert you to the affordances available for customisation or direct contributions 
+    #. to provide a map of where and when different events occur, which should aid in tracking down bugs.
+    
+
+A detailed pipeline exploration
+===============================
+
+Nbconvert takes in a notebook, which is a JSON object, and operates on that object. 
+
+This can include operations that take a notebook and return a notebook.
+For example, that operation could be to execute the notebook as though it were a continuous script; if it were executed ``--in-place`` then it would overwite the current notebook.
+Or it could be that we wish to systematically alter the notebook, for example by clearing all output cells.
+Format agnostic operations on cell content that do not violate the nbformat spec can be interpreted as a notebook to notebook conversion step; such operations can be performed as part of the preprocessing step.
```

Comment:
I'd thought through that and had decided against it. The reason I hadn't emphasised it in that way was that there is also a "notebook" exporter, so technically that is how nbconvert converts notebooks to notebooks (even if that's primarily a wrapper for preprocessors). And putting in a qualifier made it sound confusing.

Can you think of a way to say that without saying that preprocessing is the term that covers all notions of notebook to notebook conversion?